lib/time_math.rb

```diff
@@ -0,0 +1,43 @@
+require 'mechanize'
+require 'csv'
+require 'active_support'
+require 'active_support/core_ext'
+
+class TimeMath
+
+  def self.datetime_for_airport(date, time, airport_code)
+    Time.zone = airport_timezones[airport_code]
+    Time.zone.parse("#{date} #{time}")
+  end
+
+  # get the list of IATA codes for airports Southwest services
+  # cache them in a class variable, they don't change often
+  def self.swa_airports
+    agent = Mechanize.new
+    @@swa_airports ||= agent.get('https://www.southwest.com/html/air/airport-information.html')
```

Comment:
You are creating a Mecahnize instance each time this method is invoked. You can place use this same memoizing syntax and encapsulate the creation of the instance

```
@@swa_airports ||= begin
  agent = Mechanize.new
  agent.get('https://www.southwest.com/html/air/airport-information.html')
end
```